controllers/defaults/placements.go

```diff
@@ -24,8 +24,8 @@ var (
 
 		"mon": {
 			PodAntiAffinity: &corev1.PodAntiAffinity{
-				PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
-					getWeightedPodAffinityTerm(100, "rook-ceph-mon"),
+				RequiredDuringSchedulingIgnoredDuringExecution: []corev1.PodAffinityTerm{
```

Comment:
This required antiaffinity cannot be applied to mons in the stretch cluster, since two mons are created for each of the two non-arbiter zones. If it's a stretch cluster, let's completely skip adding this pod antiaffinity. Rook basically controls the zone assignment internally in that case already.

Reply:
Not adding podAntiAffinity placement for mons in case of stretched cluster. Ready for review.